lib/trino-cache/src/test/java/io/trino/cache/TestEvictableCache.java

```diff
@@ -585,4 +588,26 @@ public void testPutOnNonEmptyCacheImplementation()
                 .isInstanceOf(UnsupportedOperationException.class)
                 .hasMessage("The operation is not supported, as in inherently races with cache invalidation");
     }
+
+    @RepeatedTest(1000)
+    public void testParallelLoadingCacheEntries()
+    {
+        Cache<String, String> cache = EvictableCacheBuilder.newBuilder()
+                .expireAfterWrite(Duration.ofSeconds(60))
+                .maximumSize(10)
+                .build();
+        try (ExecutorService executor = Executors.newFixedThreadPool(2)) {
+            Runnable longRunningCacheLoad = () -> {
+                try {
+                    cache.get("key", () -> "value");
```

Comment:
Is it possible to make this always fail (or more frequently?) by using a synchronization primitive within the value loader to ensure that the loaders both run at the same time?

Reply:
good call, will try that in a second

Reply:
i cannot sync anywhere in `cache.get()`, as that locks key, and actual race happen when cache result is returned, in `finally`, looking if I can make some test-only abstraction for that